types/react-leaflet/index.d.ts

```diff
@@ -196,13 +196,13 @@ export interface TileLayerProps extends TileLayerEvents, Leaflet.TileLayerOption
     children?: Children;
     url: string;
 }
-export class TileLayer<P extends TileLayerProps, E extends Leaflet.TileLayer> extends GridLayer<P, E> { }
+export class TileLayer<P = TileLayerProps, E extends Leaflet.GridLayer = Leaflet.TileLayer> extends GridLayer<P, E> { }
```

Comment:
I agree that `E` probably should be subtype of `Leaflet.TileLayer` (same for `WMSTileLayer`, `LayersControl`, and `ZoomControl`)  So we can't do "(2) remove the constraints and just leave the default type parameters" because default type parameter doesn't provide any constraint on `E` so `E` will get infer as `{ }` and ,thus, will result in assignability error in extend clause. The reason it is not needed in the first parameter `P` in this case because all properties of `P` are options so even when `P` is inferred to be empty object it is assignable.